### PR TITLE
Migrate Water fluid properties tests to unit

### PR DESCRIPTION
**Pull request by cpgr**
- Adds unit tests of all IAPWS properties and derivatives available in `fluid_properties`
- Removes material in `fluid_properties` that was included to mimic unit testing of water properties
- Also adds headers that @andrsd forgot in #7950 (hehehe - finally found something that he missed rather than him finding all my mistakes)
- Removes a whole bunch of tests in `fluid_properties` that are now performed more extensively in unit tests

I'll follow up by moving the other fluid properties into unit tests after this

Refs #7972

### CODE REVIEW

File: unit/src/Water97FluidPropertiesTest.C [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+  catch(const std::exception & e)
+  {
+    std::string msg(e.what());
+    CPPUNIT_ASSERT( msg.find("Pressure 5.1e+07 is out of range in Water97FluidProperties::inRegion") != std::string::npos );
+  }
+
+  try
+  {
+  // Trigger invalid temperature error
+  region = _fp->inRegion(5.0e6, 2001.0);
```
Comment (andrsd) [inline unit/src/Water97FluidPropertiesTest.C]: Indent inside `try`

File: unit/src/Water97FluidPropertiesTest.C [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+  REL_TEST( "b23T", _fp->b23T(16.5291643e6), 623.15, 1.0e-8 );
+  REL_TEST( "b23p", _fp->b23p(623.15), 16.5291643e6, 1.0e-8 );
+}
+
+void
+Water97FluidPropertiesTest::pSat()
+{
+    REL_TEST( "pSat", _fp->pSat(300), 3.53658941e3, 1.0e-8 );
+    REL_TEST( "pSat", _fp->pSat(500), 2.63889776e6, 1.0e-8 );
+    REL_TEST( "pSat", _fp->pSat(600), 12.3443146e6, 1.0e-8 );
```
Comment (andrsd) [inline unit/src/Water97FluidPropertiesTest.C]: just 2 spaces...

File: unit/include/Water97FluidPropertiesTest.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+//CPPUnit includes
+#include "GuardedHelperMacros.h"
+
+class MooseMesh;
+class FEProblem;
+class Water97FluidProperties;
+
+class Water97FluidPropertiesTest : public CppUnit::TestFixture
+{
+  CPPUNIT_TEST_SUITE( Water97FluidPropertiesTest );
```
Comment (andrsd) [inline unit/include/Water97FluidPropertiesTest.h]: No spaces inside `(  ... )` and in the rest of the file.

File: unit/src/Water97FluidPropertiesTest.C [hunk 4]
```diff
@@ -0,0 +1,477 @@
...
+#include "MooseApp.h"
+#include "Utils.h"
+#include "Water97FluidPropertiesTest.h"
+
+#include "FEProblem.h"
+#include "AppFactory.h"
+#include "GeneratedMesh.h"
+#include "Water97FluidProperties.h"
+
+CPPUNIT_TEST_SUITE_REGISTRATION( Water97FluidPropertiesTest );
```
Comment (andrsd) [inline unit/src/Water97FluidPropertiesTest.C]: No spaces inside `(  ... )` and in the rest of the file.

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by andrsd** [comment]
> Also adds headers that @andrsd forgot in #7950 (hehehe - finally found something that he missed rather than him finding all my mistakes)

So you are saying that the reviewer did a sloppy job ;-)

> Removes a whole bunch of tests in fluid_properties that are now performed more extensively in unit tests

👍

File: unit/src/Water97FluidPropertiesTest.C [hunk 5]
```diff
@@ -0,0 +1,477 @@
...
+  _factory = &_app->getFactory();
+
+  registerObjects(*_factory);
+  buildObjects();
+}
+
+void
+Water97FluidPropertiesTest::tearDown()
+{
+  delete _fe_problem;
```
Comment (andrsd) [inline unit/src/Water97FluidPropertiesTest.C]: Add:
```
delete _mesh;
delete _app;
```
@dschwen in #7978 is fixing this kind of a memory leak.

**Review by andrsd — CHANGES_REQUESTED** [review]

File: unit/src/Water97FluidPropertiesTest.C (hunk 5 above)
Comment (cpgr) [inline unit/src/Water97FluidPropertiesTest.C]: Thanks. No more memory leak.

**Comment by cpgr** [comment]
Thanks @andrsd - all fixed

**Review by andrsd — APPROVED** [review]

**Pull request merged**